### PR TITLE
Writer payload params bugfix

### PR DESCRIPTION
**Pull request by david-flok**
As in #345, a regression was introduced by a recent refactoring where only payload params for a given mid should be produced but all payload params were being used instead.

### CODE REVIEW

File: src/media/writer.rs [hunk 1]
```diff
@@ -33,8 +33,14 @@ impl<'a> Writer<'a> {
...
-    pub fn payload_params(&self) -> &[PayloadParams] {
-        &self.session.codec_config
+    pub fn payload_params(&self) -> Vec<&PayloadParams> {
+        let media = self.session.media_by_mid(self.mid).unwrap();
+        let all_params = self.session.codec_config.params();
+        Vec::from_iter(
+            all_params
+                .iter()
+                .filter(|p| media.remote_pts().contains(&p.pt)),
+        )
```
Comment (algesten) [inline src/media/writer.rs:43]: Ouch, this allocation is not so nice. Depends a bit on how the API is used.

I guess the only alternative is `-> impl Iterator<Item=&PayloadParams>`. Is that too hard for the user to parse though?

The other option is to duplicate the `PayloadParams` into `Media` to have the Vec ready already. Though I'm quite keen on not doing that now that finally got the definitions centrally in `Session`.

Thoughts @k0nserv ?

Comment (k0nserv) [reply src/media/writer.rs:43]: I think your options are what we have to consider, i.e. an iterator or pre-computing the value. I can't think of a different solve to that

**Comment by david-flok** [comment]
Sorry I let this languish -- I had a pending 'review' on it that I failed to submit! Oops.

Anyway, I've made the function return an iterator now. Let me know if you'd like me to fixup the history.

File: src/media/writer.rs [hunk 2]
```diff
@@ -33,8 +33,14 @@ impl<'a> Writer<'a> {
     /// Get the configured payload parameters for the `mid` this writer is for.
     ///
     /// For the [`Writer::write()`] call, the `pt` must be set correctly.
-    pub fn payload_params(&self) -> &[PayloadParams] {
-        &self.session.codec_config
+    pub fn payload_params(&self) -> Vec<&PayloadParams> {
+        let media = self.session.media_by_mid(self.mid).unwrap();
```
[the remark this answers is not in the document]
Comment (algesten) [reply src/media/writer.rs:37]: As long as the unwrap is documented with a code comment motivating why the unwrap is ok.

**Review by algesten — CHANGES_REQUESTED** [review]
Just a code comment on that unwrap and we can go.

File: src/media/writer.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (david-flok) [reply src/media/writer.rs:41]: Great! I added comments and squashed the vector version and the iterator update.

**Review by algesten — APPROVED** [review]
Nice

**Pull request merged**